### ISSUE TITLE
[weed mount]  Cannot perceive write failure when no writable volume

### ISSUE THREAD

**Issue by paochiang** [chrislusf/seaweedfs#1765]
**Describe the bug**
when there is no writable volume server, copy file to file system mounted by `weed mount`,  no error message output below the copy command,  file's metadata is written to filer store.  This misled users and made they think that the copy command works.

**System Setup**
- List the command line to start "weed master", "weed volume", "weed filer", "weed s3", "weed mount".
``` 
# setup
root@dev-4:~/swfs# weed master -port=9333 -mdir=/tmp/master
root@dev-4:~/swfs# weed volume -port=8080 -dir=/tmp/volume -max=3 -ip=127.0.0.1 -mserver=127.0.0.1:9333
root@dev-4:~/swfs# weed filer -port=8888  -ip=127.0.0.1 -master=127.0.0.1:9333
root@dev-4:~/swfs# weed mount -filer=127.0.0.1:8888 -cacheDir=/tmp/cache001 -dir=./mnt

# shut down volume server
use `ctrl + C`

# copy a directory to ./mnt
root@dev-4:~/swfs/mnt# cp -r /root/mydir ./

```
- OS version
`Ubuntu 18.04.4 LTS`
- output of `weed version`
`version 30GB 2.13 0e99531 linux amd64`
- if using filer, show the content of `filer.toml`
`default`

**Expected behavior**
output error message  & no metadata is written to filer store

**Screenshots**
![image](https://user-images.githubusercontent.com/28077875/106119887-ab2d7500-6190-11eb-9876-044f87a21192.png)
![image](https://user-images.githubusercontent.com/28077875/106119668-6acdf700-6190-11eb-816d-715059ae1c48.png)

**Additional context**
 Actually i deploy 1 master 、 1filer、 3 volume server in k8s，and replication is `002`. 
 After shutting down a volume server,  copy a directory containing some files and subdirectories to seaweed file system mounted by `weed mount`（can not work because there are only two online volume servers).   There are some error messages below the copy command:
```
cp: failed to close './gemini-deploy/deployments/.git/FETCH_HEAD': Input/output error
cp: failed to close './gemini-deploy/deployments/.git/objects/pack/pack-9478e6d33308f94e72fd38dd453797ccd336e19c.pack': Input/output error
cp: failed to close './gemini-deploy/deployments/.git/objects/87/ecbacadd04041fcacae06b9c046a9fb38e1642': Input/output error
cp: failed to close './gemini-deploy/deployments/.git/objects/1f/5581bebc24038154af768df5a1265cf68a22c1': Input/output error
cp: failed to close './gemini-deploy/deployments/.git/objects/8a/b97b4f4bfe39da43799ff2481194ca6a670a81': Input/output error
cp: failed to close './gemini-deploy/deployments/.git/objects/5f/13ade836541df7aca38e9e124ea50e8c3e2496': Input/output error
```
 But not every file has an error message.

**Comment by chrislusf** [comment]
related to https://github.com/bazil/fuse/issues/26

**Comment by paochiang** [comment]
I don’t know much about fuse, but I found the following information, an hope this information is useful：

1、return error in libfuse :  https://github.com/libfuse/libfuse/blob/b9e3ea01dbbbba9518da216dd29c042af871ae31/include/fuse.h#L468-L496

2、In this article, it talks about how to return error： https://developer.ibm.com/articles/l-fuse/

**Comment by chrislusf** [comment]
The fix cf252fc is working. The proper error was returned with file close()
